### PR TITLE
[DCJ-89] B2C Auth base

### PR DESCRIPTION
**Pull request by sjkobori**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-89
### Summary
https://github.com/DataBiosphere/duos-ui/pull/2512 has become pretty large an hard to parse. This PR will be the base PR which all tickets from https://broadworkbench.atlassian.net/browse/DCJ-319 which are dependent on this base PR will first be merged into.

### CODE REVIEW

**Comment by sjkobori** [comment]
https://github.com/DataBiosphere/duos-ui/pull/2583 history major version bump broke the history typing. I reverted that bump in the PR for the time being.

**Comment by rushtong** [comment]
Closing this PR. Most of the work here was added in https://github.com/DataBiosphere/duos-ui/pull/2667.
